pkg/apply/merge.go

```diff
@@ -239,6 +243,16 @@ func mergeConfigMapForUpdate(current, updated *uns.Unstructured) error {
 	}
 
 	var mergedConfigMap configMapData
+
+	for i, a1 := range st1.AddressPools {
+		for j, a2 := range st2.AddressPools {
+			if a2.Name == a1.Name {
+				st1.AddressPools[i] = a2
```

Comment:
use `a2.DeepCopy` here

Reply:
why @fedepaol this flat structure copy ?

Reply:
because of the pointers fields. The updated struct is coming from the controller who owns it, and you should not modify it.
It could recycle it for efficiency for example, changing the value pointed by the pointer which is something that you don't want to